src/transforms/hexbin.js

```diff
@@ -22,10 +23,11 @@ function hexbinn(outputs, {radius = 10, ...options}) {
     ...basic(options, (data, facets) => {
       for (const o of outputs) o.initialize(data);
       return {data, facets};
-    }),
-    initialize(facets, {x: X, y: Y}, {x, y}) {
+    }, (facets, {x: X, y: Y}, scales) => {
       if (X === undefined) throw new Error("missing channel: x");
       if (Y === undefined) throw new Error("missing channel: y");
+      const x = X.scale !== undefined ? scales[X.scale] : identity.transform;
+      const y = Y.scale !== undefined ? scales[Y.scale] : identity.transform;
```

Comment:
This is fixing the behavior of the hexbin transform when identity scales are used?

Reply:
Actually, no, since we still have scale objects when position scales use the identity scale type…

So this is instead respecting the incoming channel’s scale definition, rather than assuming that the _x_ channel is bound to the _x_ scale and the _y_ channel is bound to the _y_ scale. That’s interesting and I hadn’t considered that possibility; all the built-in marks bind _x_ and _y_ to the corresponding scales, but it is theoretically possible for a custom mark not too.

Reply:
yes, that's for when you compose layouts, and the previous layout has returned channels that are already scaled. Typically hexbin + dodgeY leads to this situation.

Reply:
Aha. Thanks for the context.